Main.java

```diff
@@ -0,0 +1,153 @@
+import java.util.*;
+
+public class Main {
+
+	public static void main(String[] args) {
+
+		System.out.println("1. Example from assignment. Should print: CAR CAT CARD");
+
+		String[] d = new String[] { "CAR", "CARD", "CART", "CAT" };
+		Dictionary dict = new Dictionary(d);
+		Character [][] grid = new Character [3][2];
+		grid[0][0] = 'A';
+		grid[0][1] = 'T';
+		grid[1][0] = 'A';
+		grid[1][1] = 'C';
+		grid[2][0] = 'R';
+		grid[2][1] = 'D';
+
+		System.out.print("\tResult: ");
+		Set<String> words = findWords(grid, dict);
+		for (String w : words) {
+			System.out.print(w + " ");
+		}
+
+		System.out.println("\n2. Empty grid. Shouldn't find any word");
+
+		grid = new Character[][]{
+			{' ', ' '},
+			{' ', ' '}, 
+			{' ', ' '}
+		};
+		System.out.print("\tResult: ");
+		words = findWords(grid, dict);
+		for (String w : words) {
+			System.out.print(w + " ");
+		}
+
+		System.out.println("\n3. Empty dictionary. Shouldn't find any word");
+
+		grid[0][0] = 'A';
+		grid[0][1] = 'T';
+		grid[1][0] = 'A';
+		grid[1][1] = 'C';
+		grid[2][0] = 'R';
+		grid[2][1] = 'D';
+
+		dict = new Dictionary(new String[] {});
+
+		System.out.print("\tResult: ");
+		words = findWords(grid, dict);
+		for (String w : words) {
+			System.out.print(w + " ");
+		}
+
+		System.out.println("\n4. Empty grid and empty dictionary. Shouldn't find any word");
+
+		grid = new Character[100][100];
+		dict = new Dictionary(new String[] {});
+
+		System.out.print("\tResult: ");
+		words = findWords(grid, dict);
+		for (String w : words) {
+			System.out.print(w + " ");
+		}
+
+		System.out.println("\n5. Grid without words from dictionary. Shouldn't find any word");
+
+		grid = new Character[100][100];
+		for (int i = 0; i < 100; i++) {
+			for (int j = 0; j < 100; j++) {
+				grid[i][j] = (i + j % 2 == 0) ? 'A' : 'B';
+			}
+		}
+
+		dict = new Dictionary(new String[] { "AAAAAAAAAAAAAAAAAAAAAAAAAAAC", "ABABABABABABAAAAAAAAAAAAAAAABBBBAAAAX" });
+
+		System.out.print("\tResult: ");
+		words = findWords(grid, dict);
+		for (String w : words) {
+			System.out.print(w + " ");
+		}
+
+		System.out.println(
+				"\n5. Big grid with words from dictionary. Should find AAAAAAAAAAAAAAAAAAAAAA BBBBBBBBBBBBBBBBBBBBBBBBBBB ABABABABABABABABAB");
+
+		grid = new Character[100][100];
```

Comment:
@Thomas-de-Rivaz  Normally it will take a lot of time. However in this case it is immediate since all the words from the dictionary appear multiple times in the grid. Once all words are found it stops searching since no more words are left in the dictionary (From this test came the idea to remove the words from the dictionary).